### PR TITLE
Update docs around private github submodules in automated builds.

### PR DESCRIPTION
**Pull request by programmerq**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Note to reviewers; I created a [pull-request](https://github.com/programmerq/docker/pull/1) on @programmerq s branch to optimize the images in this PR, so wait with merging if you want those changes included in the PR.

**Comment by SvenDowideit** [comment]
LGTM - @fredlf @jamtur01 @spf13 @moxiegirl

**Comment by jamtur01** [comment]
LGTM

File: docs/sources/docker-hub/builds.md [hunk 1]
```diff
@@ -63,13 +63,15 @@ public or private GitHub repositories with a `Dockerfile`.
...
+error message in your build.
 
-Your Docker Hub deploy key is located under the "Build Details"
-menu on the Automated Build's main page in the Hub. Add this key
-to your GitHub submodule by visiting the Settings page for the
-repository on GitHub and selecting "Deploy keys".
+Normally, the Docker Hub sets up a deploy key in your GitHub repository.
+Unfortunately, GitHub only allows a deploy key to access one single repository.
+
+To work around this, we will be creating a dedicated user account in GitHub to
```
Comment (fredlf) [inline docs/sources/docker-hub/builds.md]: Please use "you" to refer to the user, and "we" to refer to Docker.

File: docs/sources/docker-hub/builds.md [hunk 2]
```diff
@@ -63,13 +63,15 @@ public or private GitHub repositories with a `Dockerfile`.
...
-need to add a deploy key from your Docker Hub repository. 
+If your GitHub repository contains links to private submodules, you'll get an
+error message in your build.
 
-Your Docker Hub deploy key is located under the "Build Details"
-menu on the Automated Build's main page in the Hub. Add this key
-to your GitHub submodule by visiting the Settings page for the
-repository on GitHub and selecting "Deploy keys".
+Normally, the Docker Hub sets up a deploy key in your GitHub repository.
+Unfortunately, GitHub only allows a deploy key to access one single repository.
```
Comment (fredlf) [inline docs/sources/docker-hub/builds.md]: "...a deploy key access to a single repository."

File: docs/sources/docker-hub/builds.md [hunk 3]
```diff
@@ -82,15 +84,33 @@ repository on GitHub and selecting "Deploy keys".
...
+      <td>First, create the new account in GitHub. It should be given read-only
+      access to the main repository and all submodules that are needed.</td>
     </tr>
     <tr>
       <td>2.</td>
-      <td><img src="/docker-hub/hub-images/github_deploy_key.png"></td>
-      <td>In your GitHub submodule's repository Settings page, add the 
-deploy key from your Docker Hub Automated Build.</td>
+      <td><img src="/docker-hub/hub-images/gh_team_members.png"></td>
+      <td>This can be accomplished by adding the account to a read-only team on
```
Comment (fredlf) [inline docs/sources/docker-hub/builds.md]: on --> in

**Comment by fredlf** [comment]
Just a few minor copy edits, otherwise LGTM. Thanks for doing this.

**Comment by thaJeztah** [comment]
probably needs a rebase, because of https://github.com/docker/docker/pull/10929 being merged

**Comment by jfrazelle** [comment]
ping @programmerq can you update

**Comment by programmerq** [comment]
I rebased this. sorry it took so long.

**Comment by jamtur01** [comment]
LGTM Ping @moxiegirl @fredlf @SvenDowideit

**Comment by SvenDowideit** [comment]
I've made you https://github.com/programmerq/docker/pull/2 which addressed @fredlf 's nits and i think also makes it more clear that you need to add a deploy key to a user to allow it to apply to more than one repo.

File: docs/sources/docker-hub/builds.md [hunk 4]
```diff
@@ -82,15 +84,33 @@ repository on GitHub and selecting "Deploy keys".
...
+    <tr>
+      <td>4.</td>
+      <td><img src="/docker-hub/hub-images/deploy_key.png"></td>
+      <td>Your automated build's deploy key is in the "Build Details" menu
+      under "Deploy keys".</td>
+    </tr>
+    <tr>
+      <td>5.</td>
+      <td><img src="/docker-hub/hub-images/gh_add_ssh_user_key.png"></td>
+      <td>In your dedicated GitHub User account, add the deploy key from your
```
Comment (sbasyal) [inline docs/sources/docker-hub/builds.md]: This "Deploy Keys" option is only visible for private GitHub repositories and not in the public ones

**Pull request merged**